xc/device.ml

```diff
@@ -608,11 +610,32 @@ let qemu_media_change ~xs device _type params =
 	if is_upstream_qemu device.frontend.domid
 	then begin
 		let cd = "ide1-cd1" in
-		let qmp_cmd =
-			if params = ""
-			then (Qmp.Command(None, Qmp.Eject (cd, Some true)))
-			else (Qmp.Command(None, Qmp.Change (cd, params, None))) in
-		qmp_write device.frontend.domid qmp_cmd
+		match params with
```

Comment:
> If we put with_qmp inside qmp_write, the socket connection will be closed after writing the command to QEMU. In some scenarios, just like CP-23007: Implement CD change using fd over QMP, all these below operations must be in one socket connection, otherwise QEMU will close the passed file descriptor of CD image.
> 1. Send file descriptor of CD using SCM_RIGHTS to QEMU socket
> 2. Send 'add-fd' QMP command 
> 3. Send 'blockdev-change-medium' QMP command

Can we make then qmp_write() and qmp_read() both receive a c parameter for the qmp connection, so it 's possible to write code like this for the above that shares the same connection c across the necessary qmp commands:
```
with_qmp domid (fun c->
 let response1 = qmp_write c domid Send_fc(etc) in
 ...
 let response2 = qmp_write c domid Send_add_fd(etc) in
 ...
 let resopnse3 = qmp_write c domid Send_blockdev_change_medium(etc) in
 ...
)
```